### PR TITLE
Add checksum for py-pylint 2.15.0, py-astroid 2.12.10, py-setuptools 62.6.0

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Confirmed all new sha256.

File: var/spack/repos/builtin/packages/py-astroid/package.py [hunk 1]
```diff
@@ -60,4 +62,5 @@ class PyAstroid(PythonPackage):
     depends_on("py-typing-extensions@3.7.4:", when="@2.7.3: ^python@:3.7", type=("build", "run"))
     depends_on("py-typing-extensions@3.10:", when="@2.8.3: ^python@:3.9", type=("build", "run"))
     depends_on("py-setuptools@17.1:", type=("build", "run"))
-    depends_on("py-setuptools@20:", type=("build", "run"), when="@2.7.3:")
+    depends_on("py-setuptools@20:", when="@2.7.3:", type=("build", "run"))
+    depends_on("py-setuptools@62.6:62", when="@2.12.10:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-astroid/package.py:70]: You'll also need to lock down the wheel version:
```
"wheel~=0.37.1"
```

File: var/spack/repos/builtin/packages/py-astroid/package.py [hunk 2]
```diff
@@ -40,12 +41,13 @@ class PyAstroid(PythonPackage):
...
     depends_on("py-lazy-object-proxy", type=("build", "run"))
     # Starting with astroid 2.3.1, astroid's dependencies were restricted
     # to a given minor version, c.f. commit e1b4e11.
     depends_on("py-lazy-object-proxy@1.4.0:1.4", when="@2.3.1:2.7.2", type=("build", "run"))
     depends_on("py-lazy-object-proxy@1.4.0:", when="@2.7.3:", type=("build", "run"))
-    depends_on("py-six", type=("build", "run"), when="@:2.7.2")
+    depends_on("py-six", when="@:2.7.2", type=("build", "run"))
     depends_on("py-six@1.12:1", when="@2.3.3:2.7.2", type=("build", "run"))
     depends_on("py-wrapt", when="@:2.2", type=("build", "run"))
     depends_on("py-wrapt@1.11:1.12", when="@2.3.3:2.8.2", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-astroid/package.py:53]: wrapt dep needs to be updated

File: var/spack/repos/builtin/packages/py-astroid/package.py [hunk 3]
```diff
@@ -60,4 +62,5 @@ class PyAstroid(PythonPackage):
     depends_on("py-typing-extensions@3.7.4:", when="@2.7.3: ^python@:3.7", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-astroid/package.py]: Shouldn't this be:
```suggestion
    depends_on("py-typing-extensions@3.7.4:", when="@2.7.3:2.8.2 ^python@:3.7", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-pylint/package.py [hunk 4]
```diff
@@ -45,31 +46,33 @@ class PyPylint(PythonPackage):
...
     depends_on("py-isort@4.2.5:5", when="@2.3.1:", type=("build", "run"))
     depends_on("py-mccabe", type=("build", "run"))
     depends_on("py-mccabe@0.6.0:0.6", when="@2.3.1:2.11", type=("build", "run"))
     depends_on("py-mccabe@0.6.0:0.7", when="@2.13:", type=("build", "run"))
     depends_on("py-pip", type=("build"))  # see https://github.com/spack/spack/issues/27075
     # depends_on('py-setuptools-scm@1.15.0:', type='build')
-    depends_on("py-setuptools-scm", type="build", when="@2.8.2")
+    depends_on("py-setuptools-scm", when="@2.8.2", type="build")
     depends_on("py-setuptools@17.1:", type="build")
+    depends_on("py-setuptools@62.6:62", when="@2.15.0:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylint/package.py]: Wheel dep needs to be locked down

File: var/spack/repos/builtin/packages/py-pylint/package.py [hunk 5]
```diff
@@ -45,31 +46,33 @@ class PyPylint(PythonPackage):
...
-    depends_on("py-astroid@2.11.6:2.11", type=("build", "run"), when="@2.14.2:")
+    depends_on("py-astroid@1.5.1:", when="@1.7:", type=("build", "run"))
+    depends_on("py-astroid@1.6:1.9", when="@1.9.4", type=("build", "run"))
+    depends_on("py-astroid@2.0:", when="@2.2.0:", type=("build", "run"))
+    depends_on("py-astroid@2.2.0:2", when="@2.3.0:2.7", type=("build", "run"))
+    depends_on("py-astroid@2.5.6:2.6", when="@2.8.0:2.10", type=("build", "run"))
+    depends_on("py-astroid@2.8.0:2.8", when="@2.11.1", type=("build", "run"))
+    depends_on("py-astroid@2.11.2:2.11", when="@2.13.5:2.13", type=("build", "run"))
+    depends_on("py-astroid@2.11.6:2.11", when="@2.14.2:2.14", type=("build", "run"))
+    depends_on("py-astroid@2.12.4:2.14", when="@2.15.0:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylint/package.py]:
```suggestion
    depends_on("py-astroid@2.12.4:2.13", when="@2.15.0:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-pylint/package.py [hunk 6]
```diff
@@ -45,31 +46,33 @@ class PyPylint(PythonPackage):
...
-    depends_on("py-tomli@1.1.0:", type=("build", "run"), when="@2.13.5: ^python@:3.10")
-    depends_on("py-tomlkit@0.10.1:", type=("build", "run"), when="@2.14.0:")
-    depends_on("py-platformdirs@2.2.0:", type=("build", "run"), when="@2.11.1:")
-    depends_on("py-typing-extensions@3.10.0:", type=("build", "run"), when="@2.11.1: ^python@:3.9")
+    depends_on("py-six", when="@1:2.3.1", type=("build", "run"))
+    depends_on("py-toml@0.7.1:", when="@2.8.2:2.12.2", type=("build", "run"))
+    depends_on("py-tomli@1.1.0:", when="@2.13.5: ^python@:3.10", type=("build", "run"))
+    depends_on("py-tomlkit@0.10.1:", when="@2.14.0:", type=("build", "run"))
+    depends_on("py-platformdirs@2.2.0:", when="@2.11.1:", type=("build", "run"))
+    depends_on("py-typing-extensions@3.10.0:", when="@2.11.1: ^python@:3.9", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylint/package.py:72]: Missing dep on colorama on windows

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylint/package.py:72]: This package would be easier to review if the deps are listed in the same order as `pyproject.toml`

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-astroid/package.py [hunk 7]
```diff
@@ -40,24 +41,31 @@ class PyAstroid(PythonPackage):
...
     depends_on("py-lazy-object-proxy@1.4.0:", when="@2.7.3:", type=("build", "run"))
-    depends_on("py-six", type=("build", "run"), when="@:2.7.2")
+    depends_on("py-six", when="@:2.7.2", type=("build", "run"))
     depends_on("py-six@1.12:1", when="@2.3.3:2.7.2", type=("build", "run"))
     depends_on("py-wrapt", when="@:2.2", type=("build", "run"))
     depends_on("py-wrapt@1.11:1.12", when="@2.3.3:2.8.2", type=("build", "run"))
     depends_on("py-wrapt@1.11:1.13", when="@2.8.3:2.10", type=("build", "run"))
-    depends_on("py-wrapt@1.11:1", when="@2.11:", type=("build", "run"))
+    depends_on("py-wrapt@1.11:1", when="@2.11", type=("build", "run"))
+    depends_on("py-wrapt@1.11:1", when="@2.12.10: ^python@:3.11", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-astroid/package.py]:
```suggestion
    depends_on("py-wrapt@1.11:1", when="@2.12.10: ^python@:3.10", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-pylint/package.py [hunk 8]
```diff
@@ -43,33 +44,37 @@ class PyPylint(PythonPackage):
     depends_on("python@3.6:", when="@2.8.2:", type=("build", "run"))
     depends_on("python@3.6.2:", when="@2.13.5:", type=("build", "run"))
     depends_on("python@3.7.2:", when="@2.14.0:", type=("build", "run"))
+    # depends_on('py-setuptools-scm@1.15.0:', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylint/package.py]: Do you know why this comment was originally added?

File: var/spack/repos/builtin/packages/py-pylint/package.py [hunk 9]
```diff
@@ -43,33 +44,37 @@ class PyPylint(PythonPackage):
...
     depends_on("py-mccabe@0.6.0:0.6", when="@2.3.1:2.11", type=("build", "run"))
     depends_on("py-mccabe@0.6.0:0.7", when="@2.13:", type=("build", "run"))
+    depends_on("py-tomli@1.1.0:", when="@2.13.5: ^python@:3.10", type=("build", "run"))
+    depends_on("py-tomlkit@0.10.1:", when="@2.14.0:", type=("build", "run"))
+    depends_on("py-colorama@0.4.5:", when="platform=windows", type=("build", "run"))
+    depends_on("py-typing-extensions@3.10.0:", when="@2.11.1: ^python@:3.9", type=("build", "run"))
+    depends_on("py-backports-functools-lru-cache", when="^python@:2.8", type=("build", "run"))
+    depends_on("py-configparser", when="^python@:2.8", type=("build", "run"))
+    depends_on("py-editdistance", when="@:1.7", type=("build", "run"))
     depends_on("py-pip", type=("build"))  # see https://github.com/spack/spack/issues/27075
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pylint/package.py]: You can actually remove pip, it's in the base class now

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pylint/package.py (hunk 8 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-pylint/package.py]: No idea.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-pylint/package.py]: Looks like it's been around for 5+ years, you can just remove it

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
@scottwittenburg any idea about gitlab failure?

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Pull request closed without merging**